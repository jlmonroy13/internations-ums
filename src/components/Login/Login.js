import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import AccountFormField from '../AccountFormField/';

class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      errorMessage: '',
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }
  handleSubmit(event) {
    event.preventDefault();
    const { email, password } = this.state;
    this.props.login({ email, password }).then((response) => {
      if (response.message) {
        this.setState(() => ({
          errorMessage: response.message,
        }));
      }
    });
  }
  handleChange(event) {
    const { name, value } = event.target;
    this.setState(() => ({
      [name]: value,
      errorMessage: '',
    }));
  }
  render() {
    const { email, password, errorMessage } = this.state;
    return (
      <div className="grid__item medium--one-half">
        <section className="account-box">
          <h1 className="account-box__title">Log in</h1>
          <form onSubmit={event => this.handleSubmit(event)}>
            <div className="push-triple--bottom relative">
              <AccountFormField
                placeholder="Email"
                id="email"
                name="email"
                value={email}
                iconName="user"
                type="email"
                required={true}
                onChange={event => this.handleChange(event)}
              />
              <AccountFormField
                placeholder="Password"
                id="password"
                name="password"
                value={password}
                iconName="lock"
                type="password"
                required={true}
                onChange={event => this.handleChange(event)}
              />
              <div className="account-box__error-message">{errorMessage}</div>
            </div>
            <button className="button button--submit" type="submit">
              Log in
            </button>
          </form>
        </section>
        <div className="account__message">
          {"Don't have an account?"} <Link to="/signup">Create an account</Link>
        </div>
      </div>
    );
  }
}

Login.propTypes = {
  login: PropTypes.func.isRequired,
};

export default Login;
