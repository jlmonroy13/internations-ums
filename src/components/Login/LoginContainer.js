import { connect } from 'react-redux';
import { login } from '../../actions/account';
import Login from './Login';

function mapDispatchToProps(dispatch) {
  return {
    login: data => dispatch(login(data)),
  };
}

export default connect(null, mapDispatchToProps)(Login);
