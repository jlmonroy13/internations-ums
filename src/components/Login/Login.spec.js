import React from 'react';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('<Login />', () => {
  it('should render the login component', () => {
    const login = jest.fn();
    const wrapper = renderer.create(<Login login={login} />);
    expect(wrapper).toMatchSnapshot();
  });
});
