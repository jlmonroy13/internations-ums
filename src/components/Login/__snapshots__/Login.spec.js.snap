// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render the login component 1`] = `
<div
  className="grid__item medium--one-half"
>
  <section
    className="account-box"
  >
    <h1
      className="account-box__title"
    >
      Log in
    </h1>
    <form
      onSubmit={[Function]}
    >
      <div
        className="push-triple--bottom relative"
      >
        <div
          className="account__form-group"
        >
          <label
            className="account__label account__label--with-icon"
            htmlFor="email"
          >
            <span
              className="icon icon--user"
            />
          </label>
          <input
            className="account__form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="account__form-group"
        >
          <label
            className="account__label account__label--with-icon"
            htmlFor="password"
          >
            <span
              className="icon icon--lock"
            />
          </label>
          <input
            className="account__form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="account-box__error-message"
        >
          
        </div>
      </div>
      <button
        className="button button--submit"
        type="submit"
      >
        Log in
      </button>
    </form>
  </section>
  <div
    className="account__message"
  >
    Don't have an account?
     
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Create an account
    </a>
  </div>
</div>
`;
